### DISCUSSION TITLE
Is there a way to make scrolling send arrow keys in alternate screen mode?

### DISCUSSION THREAD

**Discussion by slmjkdbtl** [wezterm/wezterm#1194] [Q&A] [answered]
Similar to the "Scroll wheel sends arrow keys when in alternate screen mode" in iTerm2's advance settings.

**Comment by wez** [comment, marked as answer]
It does by default, but if you are using an application that has enabled mouse reporting, then we have to send the mouse movement to the app instead of the arrow keys.

**Reply by slmjkdbtl** [reply]
Oh I see. Thanks for the reply. Do you think it's ok to have an option to still send arrow keys on wheel when mouse report is enabled? Or there's actually some conflict

**Reply by wez** [reply]
It conflicts with the defined behavior of mouse reporting mode.  What problem are you trying to solve here?  It sounds like the application you're using in the terminal might need to be configured differently!

**Reply by slmjkdbtl** [reply]
I'm migrating my current iTerm configs to WezTerm, and the wheel as arrow + mouse experience is something I really hope to get 😢 

So I'm trying out a [fork](https://github.com/slmjkdbtl/wezterm/commit/20ec0494d7947210fd7672a78f0adc2c0a275ead) where I added an option to always send arrow keys on wheel in alternate screen mode, it works but it looks like it's still scrolled by normal wheel behavior somehow 🤔 not sure where the problem is.

It's a joy to build and run the project btw! The code looks clear and everything just works.

**Reply by wez** [reply]
What I'd like to understand is specifically which application you are running that enables alt screen and mouse reporting and that doesn't support processing the scroll wheel reports?  I'd like to understand that, as it is the heart of your question and will inform the correct solution!

**Reply by slmjkdbtl** [reply]
Vim. It supports scroll wheel events, but just personal preference like to act as arrow keys instead

**Reply by slmjkdbtl** [reply]
Actually never mind. I just `map <ScrollWheelUp> <up>` in vim and it works perfectly, no need to an additional flag.
